dashboard/app/assets/javascripts/progress.js

```diff
@@ -76,15 +76,19 @@ window.dashboard.progress = (function () {
     }
   };
 
-  progress.renderStageProgress = function (stageData, progressData, clientProgress, currentLevelId) {
+  progress.renderStageProgress = function (stageData, progressData, clientProgress, currentLevelId, puzzlePage) {
     var serverProgress = progressData.levels || {};
     var currentLevelIndex = null;
 
-    var combinedProgress = stageData.levels.map(function(level, index) {
-      if (level.id === currentLevelId) {
-        currentLevelIndex = index;
-      }
+    // Determine currentLevelIndex from the URL's /puzzle/XX value.
+    var parsedLevelIndex = window.location.pathname.match(/\/puzzle\/(\d+)/);
```

Comment:
hm, maybe at this point we should just pass currentLevelIndex in progressData or something. We do have URLs that do not contain "puzzle" (eg. https://studio.code.org/hoc/1), does that still work?